static/js/util/profile_edit.js

```diff
@@ -191,37 +193,42 @@ export function boundDateField(keySet: string[], label: string, omitDay: boolean
       day: day !== undefined ? day : edit.day
     });
 
-    // these functions return undefined if a month, day, or year is invalid, and converts the value to a number
-    // except if the input value is an empty string, in which case an empty string is returned
-    let validatedDay = 1;
+    let str = S.maybe("", String); // returns String(Maybe.value) || ""
+
+    let validatedDay = Maybe.of(1);
     if (!omitDay) {
       validatedDay = validateDay(newEdit.day);
-      newEdit.day = validatedDay === undefined ? '' : String(validatedDay);
+      newEdit.day = str(validatedDay);
     }
 
     let validatedMonth = validateMonth(newEdit.month);
-    newEdit.month = validatedMonth === undefined ? '' : String(validatedMonth);
+    newEdit.month = str(validatedMonth);
 
     let validatedYear = validateYear(newEdit.year);
-    newEdit.year = validatedYear === undefined ? '' : String(validatedYear);
+    newEdit.year = str(validatedYear);
 
     // keep text up to date
     _.set(clone, editKeySet, newEdit);
 
-    if ([validatedDay, validatedMonth, validatedYear].includes(undefined)) {
-      // store the edit value and make the formatted date undefined so it doesn't pass validation
+    let padYear = s => _.padStart(s, 4, '0');
+
+    let validCheck = dates => Maybe.of(R.all(S.isJust)(dates)).filter(R.equals(true));
```

Comment:
I feel like `validCheck` should be renamed `allJust` here, since there's nothing date specific about it. I would also add a comment: `allJust returns Just(true) if all items are of type Just, or else Nothing`